plugins/module_utils/job.py

```diff
@@ -122,430 +114,190 @@ def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
-    Say 'subsystem'||':'||value('ESYSID'||"."||ix)
-    Say 'system'||':'||value('SYSNAME'||"."||ix)
-    Say 'owner'||':'||value('OWNERID'||"."||ix)
-    Say 'ret_code_msg'||':'||value('RETCODE'||"."||ix)
-    Say 'class'||':'||value('JCLASS'||"."||ix)
-    Say 'content_type'||':'||value('JTYPE'||"."||ix)
-    Address SDSF "ISFACT ST TOKEN('"TOKEN.ix"') PARM(NP ?)",
-"("prefix JDS_
-    lrc=rc
-    if lrc<>0 | JDS_DDNAME.0 == 0 then do
-    Say '-----NO DD NAMES FOUND-----'
-    end
-    else do
-    Say '-----START OF DD NAMES-----'
-    do jx=1 to JDS_DDNAME.0
-        if ddname == '' | ddname == value('JDS_DDNAME'||"."||jx) then do
-        Say '-----START OF DD-----'
-        Say 'ddname'||':'||value('JDS_DDNAME'||"."||jx)
-        Say 'record_count'||':'||value('JDS_RECCNT'||"."||jx)
-        Say 'id'||':'||value('JDS_DSID'||"."||jx)
-        Say 'stepname'||':'||value('JDS_STEPN'||"."||jx)
-        Say 'procstep'||':'||value('JDS_PROCS'||"."||jx)
-        Say 'byte_count'||':'||value('JDS_BYTECNT'||"."||jx)
-        Say '-----START OF CONTENT-----'
-        Address SDSF "ISFBROWSE ST TOKEN('"token.ix"')"
-        do kx=1 to isfline.0
-            Say isfline.kx
-        end
-        Say '-----END OF CONTENT-----'
-        Say '-----END OF DD-----'
-        end
-    end
-    Say '-----END OF DD NAMES-----'
-    end
-    Say '-----END OF JOB-----'
-end
-end
-
-rc=isfcalls('OFF')
-
-return 0
-"""
-    try:
-        module = AnsibleModuleHelper(argument_spec={})
-        if dd_name is None or dd_name == "?":
-            dd_name = ""
-        jobid_param = "jobid=" + job_id
-        owner_param = "owner=" + owner
-        jobname_param = "jobname=" + job_name
-        ddname_param = "ddname=" + dd_name
-
-        tmp = NamedTemporaryFile(delete=True)
-        with open(tmp.name, "w") as f:
-            f.write(get_job_detail_rexx)
-        chmod(tmp.name, S_IEXEC | S_IREAD | S_IWRITE)
-        args = [jobid_param, owner_param, jobname_param, ddname_param]
-
-        cmd = [tmp.name, " ".join(args)]
-        rc, out, err = module.run_command(args=cmd)
-    except Exception:
-        raise
-    return rc, out, err
-
-
-def job_status(job_id=None, owner=None, job_name=None):
+def job_status(job_id=None, owner=None, job_name=None, dd_name=None):
     """Get the status information of a z/OS job based on various search criteria.
 
     Keyword Arguments:
         job_id {str} -- The job ID to search for (default: {None})
         owner {str} -- The owner of the job (default: {None})
         job_name {str} -- The job name search for (default: {None})
+        dd_name {str} -- If populated, return ONLY this DD in the job list (default: {None})
 
     Returns:
         list[dict] -- The status information for a list of jobs matching search criteria.
         If no job status is found, this will return an empty job code with msg=JOB NOT FOUND
-
+        new format: Job(owner=job[0], name=job[1], id=job[2], status=job[3], rc=job[4]))
```

Comment:
Is this a left over comment, trying to understand how it relates to what is returned; sort of seems like you are documenting the dictionary returned but also looks like it was a development note.

Reply:
Yep... I'll re-push with that comment removed (there were too many things called job or jobs, so I left myself that note and forget to yank it.)